CRM/Contact/Page/View/Relationship.php

```diff
@@ -217,17 +230,8 @@ public function run() {
   }
 
   public function setContext() {
-    $context = CRM_Utils_Request::retrieve('context', 'Alphanumeric',
-      $this, FALSE, 'search'
-    );
-
-    if ($context == 'dashboard') {
-      $cid = CRM_Utils_Request::retrieve('cid', 'Integer',
-        $this, FALSE
-      );
-      $url = CRM_Utils_System::url('civicrm/user',
-        "reset=1&id={$cid}"
-      );
+    if ($this->getContext() == 'dashboard') {
+      $url = CRM_Utils_System::url('civicrm/user', "reset=1&id={$this->getContactId()}");
```

Comment:
I don't know if a function call inside curly places in a quote is valid php - I'm guessing maybe it is - but stylistically we have no precedent for it

Reply:
I think it's new to php5, but it works fine; I've done it myself in some spots.

https://www.php.net/manual/en/language.types.string.php#language.types.string.parsing.complex

Reply:
ok - I guess it's ok then - will merge